libcxx/include/__vector/vector.h

```diff
@@ -84,7 +84,7 @@ _LIBCPP_PUSH_MACROS
 _LIBCPP_BEGIN_NAMESPACE_STD
 
 template <class _Tp, class _Allocator /* = allocator<_Tp> */>
-class _LIBCPP_TEMPLATE_VIS vector {
+class _LIBCPP_TEMPLATE_VIS vector _LIBCPP_MAYBE_FORCE_NONSTANDARD_LAYOUT {
```

Comment:
`std::vector` is used in the libc++ dylib across the ABI boundary. So this means you can only use `pointer_field_protection` when the dylib is build with `pointer_field_protection`.

Reply:
The intent would be that the user builds libc++.a with `pointer_field_protection` and statically links it into their binary. Because of the changes to pointer representation there are numerous other ABI incompatibilities. This feature is incompatible with dynamic linking so we may want to skip building the dynamic library if the feature is enabled.